src/constraints/StringConstraints.ts

```diff
@@ -3,7 +3,17 @@ import { Result } from '../lib/Result';
 import type { IConstraint } from './base/IConstraint';
 import { Comparator, eq, ge, gt, le, lt, ne } from './util/operators';
 
-export type StringConstraintName = `s.string.length${'Lt' | 'Le' | 'Gt' | 'Ge' | 'Eq' | 'Ne'}`;
+export type StringConstraintName = `s.string.${`length${'Lt' | 'Le' | 'Gt' | 'Ge' | 'Eq' | 'Ne'}` | 'regex' | 'url' | 'uuid' | 'email'}`;
+export type StringProtocol = `${string}:`;
+export type StringDomain = `${string}.${string}`;
+
+export interface UrlOptions {
```

Comment:
Those aren't exported, check for where `StringContraintName` is exported at.

Reply:
wdym? I didn't understand.

Reply:
We can't do `import type { UrlOptions, StringProtocol, StringDomain } from '@sapphire/shapeshift';`, which is something we'd like to be able to since they're public-facing.

Reply:
Thank u for the detailed explanation. Fixed